api/replica.go

```diff
@@ -50,3 +54,12 @@ func (p *Replica) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		httpError(w, fmt.Sprintf("Unable to reach primary cluster manager (%s): %v", err, p.primary), http.StatusInternalServerError)
 	}
 }
+
+func (p *Replica) ping(w http.ResponseWriter, r *http.Request) {
+	if p.primary == "" {
+		httpError(w, "No elected primary cluster manager", http.StatusInternalServerError)
+		w.Write([]byte{'E', 'R', 'R', 'O', 'R'})
```

Comment:
Heh. I wanted to maintain consistency with this:

https://github.com/docker/swarm/blob/master/api/handlers.go#L963

But I'll go ahead and change it to your version.